iotedgehubdev/certutils.py

```diff
@@ -227,6 +227,100 @@ def get_cert_file_path(id_str, dir_path):
     def get_pfx_file_path(id_str, dir_path):
         return os.path.join(dir_path, id_str, 'cert', id_str + EC.PFX_SUFFIX)
 
+    def load_cert_from_file(self, id_str, cert_path, key_path, key_passphrase):
+        if id_str in list(self._cert_chain.keys()):
+            msg = 'Could not load cert from file. Certificate already in chain. ID: {0}'.format(id_str)
+            raise EdgeValueError(msg)
+        cert_dict = {}
+        with open(cert_path, 'r') as cert_file:
+            cert_content = cert_file.read()
+            cert_dict['cert'] = crypto.load_certificate(crypto.FILETYPE_PEM, cert_content)
+        with open(key_path, 'r') as key_file:
+            key_content = key_file.read()
+            try:
+                cert_dict['key_pair'] = crypto.load_privatekey(crypto.FILETYPE_PEM, key_content, key_passphrase)
+            except Exception:
+                raise EdgeInvalidArgument('Failed to load privake key. Please check your passphase.')
+        self._cert_chain[id_str] = cert_dict
+
+    def check_cert_existence(self, id_str):
+        return id_str in list(self._cert_chain.keys())
+
+    def get_cert_dict(self, id_str):
+        if id_str not in list(self._cert_chain.keys()):
+            msg = 'Certificate not in chain. ID: {0}'.format(id_str)
+            raise EdgeValueError(msg)
+        return self._cert_chain[id_str]
+
+    def dump_cert(self, id_str, cert_output_path, key_output_path, overwrite_existing):
+        self.dump_cert_content(id_str, cert_output_path, overwrite_existing)
+        self.dump_cert_key(id_str, key_output_path, overwrite_existing)
+
+    def dump_cert_content(self, id_str, output_path, overwrite_existing):
+        if not overwrite_existing and os.path.exists(output_path):
+            raise EdgeFileAccessError(
+                'Following cert file already exists.', output_path)
+        Utils.mkdir_if_needed(os.path.dirname(output_path))
+        cert_dict = self.get_cert_dict(id_str)
+        cert_obj = cert_dict['cert']
+        try:
+            with open(output_path, 'w') as output_file:
+                output_file.write(crypto.dump_certificate(crypto.FILETYPE_PEM,
+                                                          cert_obj).decode('utf-8'))
+        except IOError as ex:
+            msg = 'IO Error when exporting certs for ID: {0}.\n' \
+                  ' Error seen when exporting file {1}.' \
+                  ' Errno: {2} Error: {3}'.format(id_str, ex.filename, str(ex.errno), ex.strerror)
+            raise EdgeFileAccessError(msg, output_path)
+
+    def dump_cert_key(self, id_str, output_path, overwrite_existing):
```

Comment:
can we use  export_cert_artifacts_to_dir?

Reply:
It's hard to customize the output path in export_cert_artifacts_to_dir(). I plans to update this function later to use the dump_cert function I added.